src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java

```diff
@@ -460,12 +490,37 @@ protected Object getHandler(String serviceName) {
 	 */
 	private JsonNode invoke(Object target, Method method, List<JsonNode> params) throws IOException, IllegalAccessException, InvocationTargetException {
 		logger.debug("Invoking method: {} with args {}", method.getName(), params);
-		Object[] convertedParams = convertJsonToParameters(method, params);
-		if (convertedParameterTransformer != null) {
-			convertedParams = convertedParameterTransformer.transformConvertedParameters(target, convertedParams);
-		}
-		Object result = method.invoke(target, convertedParams);
+
+		Object[] convertedParams;
+		Object result;
+
+        if(method.getGenericParameterTypes().length==1 && method.isVarArgs()) {
+            convertedParams = new Object[params.size()];
+            ObjectMapper mapper = new ObjectMapper();
+
+            for (int i = 0; i < params.size(); i++) {
+                JsonNode jsonNode = params.get(i);
+                Class type = getJavaTypeForJsonType(jsonNode.getNodeType());
+                Object object = mapper.convertValue(jsonNode, type);
+                logger.debug(String.format(
+                        "[%s] param: %s -> %s",
+                        method.getName(), i, type.getName()
+                ));
+                convertedParams[i] = object;
+            }
+
+            result = method.invoke(target, new Object[] {convertedParams});
+
+        } else {
+            convertedParams = convertJsonToParameters(method, params);
+			if (convertedParameterTransformer != null) {
```

Comment:
this can be a single line if, no parenthesis needed, also no empty line needed afterwards